### PR TITLE
Ability to switch AV and SA for all users on/off. Fixes #69

### LINKED ISSUE #69: Allow enabling spamassassin and anti-virus for all domain users
We have disabling already, now we should allow enabling it globally as well. See #66.

### PR DESCRIPTION
**Pull request by Kami-no**
Ability for domain admin to force AV and SA for all users to enable or
disable. Fixes #69

### CODE REVIEW

**Comment by rimas-kudelis** [comment]
@Kami-no would you please make a PR without whitespace changes, or at least split it into two commits? Thanks.

Because right now the true essence of this PR can hardly be seen among numerous whitespace and line-break adjustments.

**Comment by Kami-no** [comment]
I'll separate sitechange.php but sitechangesubmit.php modified almost completely for 'case' logic.

File: vexim/sitechange.php [hunk 1]
```diff
@@ -1,86 +1,153 @@
...
+                        $query = "SELECT localpart,domain FROM users,domains
+                            WHERE domains.domain_id=:domain_id
+                            AND admin=1 AND users.domain_id=domains.domain_id";
+                        $sth = $dbh->prepare($query);
+                        $sth->execute(array(':domain_id' => $_GET['domain_id']));
+                        while ($row = $sth->fetch()) {
+                            print '<option value="' . $row['localpart'] . '">' . $row['localpart'] . '</option>' . "\n\t";
+                        }
+                        ?>
+                    </select>
```
Comment (rimas-kudelis) [inline vexim/sitechange.php]: The `@` character is still missing, I suppose it's not on purpose?

Comment (rimas-kudelis) [inline vexim/sitechange.php]: Actually, you probably just shouldn't change this particular line.

**Comment by Udera** [comment]
I can't disable anti-virus for all users without touching spamassassin.

Settings for the `maximum message size` are ignored (they are updated, but the user can still choose a larger value). I am not sure, if this issue existed before.

After a rewrite of the whole script, it's probably a good idea to verify if all is working as it should: https://github.com/vexim/vexim2/blob/master/docs/checklist.txt

**Comment by Udera** [comment]
Ok, I found the problem with the `maxmsgsize`, it was independent from this pull request. So don't bother for it (@Kami-no), I created a separate pull request: https://github.com/vexim/vexim2/pull/101

It remains the issue with disabling antivirus for all users (without touching spamassassin). Problem should be here: https://github.com/vexim/vexim2/pull/100/files#diff-6449e75348218b3427547d3aafb7a947R28

**Comment by Udera** [comment]
Looks good but I didn't test again.

**Comment by Udera** [comment]
I did test it. It seems to do what it should do. 

* I would change the position of the lower submit button: it should be on the right hand side. Can you make a little spacing that it gets clearer that you need to push the second submit button? (perhaps like on the top, insert a new subheader "Modify SA/AV setting for all accounts".

* If you submit the lower form, there is no field `domain`, so the variable `$_POST['domain']` in https://github.com/Kami-no/vexim2/blob/master/vexim/sitechangesubmit.php#L124 is unknown. Either add a hidden field in the form or get the domain from the db-query.

* If spamassassin is enabled and we disable it, shouldn't we disable it automatically for all accounts?

**Comment by Udera** [comment]
I fixed some stuff and created a new pull request: https://github.com/vexim/vexim2/pull/123

**Pull request closed without merging**